lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -287,10 +316,41 @@ def log_write_failure(request, error)
         :error_class=>error.class.to_s, :error=>error.to_s
     end
 
-    def fetch_metadata(metadata_path)
-      # Fetch GCE metadata - see https://cloud.google.com/compute/docs/metadata
-      open('http://169.254.169.254/computeMetadata/v1/' + metadata_path,
-           {'Metadata-Flavor' => 'Google'}) do |f|
+    # "enum" of Platform values
+    module Platform
+      OTHER = 0  # Other/unkown platform
+      GCE = 1    # Google Compute Engine
+    end
+
+    # Determine what platform we are running on by consulting the metadata
+    # service (unless the user has explicitly disabled using that).
+    def detect_platform
+      if !@use_metadata_service
+        $log.info "use_metadata_service is false; not detecting platform"
+        return Platform::OTHER
+      end
+
+      begin
+        open('http://' + METADATA_SERVICE_ADDR) do |f|
+          if (f.meta['metadata-flavor'] == 'Google')
+            $log.info 'Detected GCE platform'
+            return Platform::GCE
+          end
+        end
+      rescue Exception => e
+        $log.debug "Failed to access metadata service: ", :error=>e
```

Comment:
fixed (I also fixed the one other place with no spaces, which was also a log message).  I have a TODO to find a linter to use and reformat things.